domain/src/main/java/org/oppia/android/domain/devoptions/ShowAllHintsAndSolutionHelper.kt

```diff
@@ -0,0 +1,19 @@
+package org.oppia.android.domain.devoptions
+
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Monitor to identify whether all hints and solution should be shown. */
```

Comment:
Please detail in this KDoc that this controller is only expected to be interacted with on the main thread.

FWIW if it isn't, then you update it to be threads safe & mention that in its KDoc.

Reply:
Updated the Kdoc as this is expected to be called on the main thread only.